File: tiq-e/array/1/main.ts
import {performance} from "perf_hooks";

function removeDuplicates(nums: number[]): number {
    if (nums.length < 2) {
        return nums.length;
    }

    let u = 0;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[u]) {
            u += 1;
            [nums[u], nums[i]] = [nums[i], nums[u]];
        }
    }

    return u + 1;
}

const nums = [1, 1, 2];

const start = performance.now();
const result = removeDuplicates(nums);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/array/1/main.go
package main

import (
	"fmt"
	"time"
)

func removeDuplicates(nums []int) int {
	if len(nums) < 2 {
		return len(nums)
	}

	u := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[u] {
			u += 1
			nums[u] = nums[i]
		}
	}

	return u + 1
}

func main() {
	var nums = []int{1, 1, 2}

	var start = time.Now()
	var result = removeDuplicates(nums)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
